From a9cc848ce01fef70384bc986cf774b3bac462213 Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Wed, 19 Oct 2011 23:47:38 +0200
Subject: [PATCH 03/18] migration: Check that migration is active before
 cancel it

RH-Author: Juan Quintela <quintela@redhat.com>
Message-id: <cb25971d65c5dec3bdb61d3765e63d01b1a9d655.1319066770.git.quintela@redhat.com>
Patchwork-id: 34428
O-Subject: [PATCH qemu-kvm RHEL-6.2 03/16] migration: Check that migration is active before cancel it
Bugzilla: 669581
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>

Signed-off-by: Juan Quintela <quintela@redhat.com>
Reviewed-by: Anthony Liguori <aliguori@us.ibm.com>

Conflicts:

	migration.c

Printing of timing conflict with this.
---
 migration.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 migration.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/migration.c b/migration.c
index d40005d..4913557 100644
--- a/migration.c
+++ b/migration.c
@@ -148,8 +148,9 @@ int do_migrate_cancel(Monitor *mon, const QDict *qdict, QObject **ret_data)
 {
     MigrationState *s = current_migration;
 
-    if (s)
+    if (s && s->get_status(s) == MIG_STATE_ACTIVE) {
         s->cancel(s);
+    }
 
     STOP_MIGRATION_CLOCK();
     DPRINTF("canceled after %lu milliseconds\n", stop);
-- 
1.7.4.4

